animation-glib/glide/glide.cpp

```diff
@@ -0,0 +1,404 @@
+{
+  return LookupTypedInterfaceProp <at::TransformAnimation, ag::GlideAnimation> (G_OBJECT (animation))->XAxisLocationUnit ();
+}
+
+void
+animation_glide_animation_set_y_axis_location_unit (AnimationGlideAnimation *animation,
+                                                    float                    y_axis_location_unit)
+{
+  LookupTypedInterfaceProp <at::TransformAnimation, ag::GlideAnimation> (G_OBJECT (animation))->YAxisLocationUnit (y_axis_location_unit);
+}
+
+float
+animation_glide_animation_get_y_axis_location_unit (AnimationGlideAnimation *animation)
+{
+  return LookupTypedInterfaceProp <at::TransformAnimation, ag::GlideAnimation> (G_OBJECT (animation))->YAxisLocationUnit ();
+}
+
+void
+animation_glide_animation_set_stepper (AnimationGlideAnimation *animation,
+                                       AnimationStepper        *stepper)
+{
+  animation::stepper::Stepper *stepper_ptr = nullptr;
+  g_object_get (stepper, "stepper", (gpointer) &stepper_ptr, NULL);
+
+  LookupTypedInterfaceProp <at::TransformAnimation, ag::GlideAnimation> (G_OBJECT (animation))->Stepper (*stepper_ptr);
+}
+
+/**
+ * animation_glide_animation_get_stepper:
+ * @animation: An #AnimationGlideAnimation
+ *
+ * Returns: (transfer full): Get the stepper for this #AnimationGlideAnimation
+ */
+AnimationStepper *
+animation_glide_animation_get_stepper (AnimationGlideAnimation *animation)
+{
+  auto const &stepper (LookupTypedInterfaceProp <at::TransformAnimation, ag::GlideAnimation> (G_OBJECT (animation))->Stepper ());
+
+  return animation_stepper_wrapper_new ((gpointer) &stepper);
+}
+
+static void
+animation_glide_animation_set_property (GObject      *object,
+                                        guint         prop_id,
+                                        const GValue *value,
+                                        GParamSpec   *pspec)
+{
+  AnimationGlideAnimation *glide_animation = ANIMATION_GLIDE_ANIMATION (object);
+
+  switch (prop_id)
+    {
+    case PROP_INITIAL_DISTANCE:
+      animation_glide_animation_set_initial_distance (glide_animation, g_value_get_float (value));
+      break;
+    case PROP_X_ROTATION_ANGLE_DEGREES:
+      animation_glide_animation_set_x_rotation_angle_degrees (glide_animation, g_value_get_float (value));
+      break;
+    case PROP_Y_ROTATION_ANGLE_DEGREES:
+      animation_glide_animation_set_y_rotation_angle_degrees (glide_animation, g_value_get_float (value));
+      break;
+    case PROP_X_AXIS_LOCATION_UNIT:
+      animation_glide_animation_set_x_axis_location_unit (glide_animation, g_value_get_float (value));
+      break;
+    case PROP_Y_AXIS_LOCATION_UNIT:
+      animation_glide_animation_set_y_axis_location_unit (glide_animation, g_value_get_float (value));
+      break;
+    case PROP_SCREEN_WIDTH:
+      /* Not writable, except on construction */
+      break;
+    case PROP_TARGET:
+      /* Not writable, except on construction */
+      break;
```

Comment:
Even on construction this would be invoked though, no?

Reply:
It is, yeah.

Its one of those weird things where we need to keep it as a construct property in order to pass it to the underlying c++ class constructor but not as a writable thing.

Tbh though they should probably both be writable all the way through - the target position or screen width could very well change during the course of the animation and we should probably know about that...